vere/main.c

```diff
@@ -544,11 +544,22 @@ main(c3_i   argc,
   printf("~\n");
   //  printf("welcome.\n");
   printf("urbit %s\n", URBIT_VERSION);
-  printf("urbit: home is %s\n", u3_Host.dir_c);
+  int mprint_i = 1000;
+  c3_c* abs_c = (c3_c *)malloc(mprint_i);
+
+  // allocates more memory as needed if the path is too large
+  //
+  while ( abs_c != getcwd(abs_c, mprint_i) ) {
+    free(abs_c);
+    mprint_i *= 2;
+    abs_c = (c3_c *)malloc(mprint_i);
+  }
+  printf("boot: home is %s/%s\n", realpath(abs_c, NULL), u3_Host.dir_c);
```

Comment:
Ok, good point. Multiple slashes or something isn't a huge concern -- you can drop `realpath()` and leave this as it was. Please use `c3_malloc()` instead of `malloc()`, and note that you don't need to cast the inner malloc call.

Reply:
That would work!

Couple style points: `c3_i` in place of `int`, no need to cast either `c3_malloc()` call, and you only need one printf/free outside the conditional.

Reply:
What do you mean "outside the conditional"? I only have one printf for "true" and one slightly different printf for "false"

Reply:
I'm doing too many things at once, this looks good. (I was picturing dir_c copied into abs_c)

Reply:
Cool! Thanks a lot for your help